File: lbo-sdk/src/plugin/cms/models/Lobby.ts
import LobbyItem from './LobbyItem'
import LobbyNavItem from "@/plugin/cms/models/LobbyNavItem";
import Badge from "@/plugin/cms/models/Badge";
import User from "@/plugin/cms/models/User";

export default class Lobby {

  id: number = 0;
  name: string;
  description: string = '';
  version: number = 1;
  active: boolean = false;
  modifiedDate: Date | null = null;
  modifiedBy: User = new User();
  lobbyItems: LobbyItem[] = [];
  channel: string;
  nav: LobbyNavItem[] = [new LobbyNavItem()];
  badges: Badge[] = [];

  public constructor(name: string, channel: string) {
    this.name = name;
    this.channel = channel;
  }

  get status(): string {
    return this.active ? 'Active' : 'Inactive'
  }

  addLobbyItem(item: LobbyItem, channel: string) {
    item.nav = this.nav;
    item.badges = this.badges;
    item.page.primary_nav_code = this.nav[0].primary_nav_code;
    item.page.channel = channel;
    item.state = "logged_in";
    item.name = this.name
    this.lobbyItems.push(item);
  }

}

File: service-casino/service-casino-cms/src/main/java/lithium/service/casino/cms/api/schema/lobby/v1/Lobby.java
package lithium.service.casino.cms.api.schema.lobby.v1;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Date;
import java.util.List;

@Data
@Builder
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Lobby {

	private Long id;

	private String name;

	private String description;

	private User createdBy;

	private Date createdDate;

	private User modifiedBy;

	private Date modifiedDate;

	private Integer version;

	private List<LobbyItem> lobbyItems;

}
